### ISSUE TITLE
Reuse endpoint map to create zone ingress (and egress?) endpoint map

### ISSUE THREAD

**Issue by jakubdyszkiewicz** [kumahq/kuma#10790]
### Description

Follow up of https://github.com/kumahq/kuma/pull/10748

> We could also mark Endpoints that target ingresses.
Then take the results of BuildEdsEndpointMap and BuildCrossMeshEndpointMap and filter out those endpoints. I didn't want to do this now, I worry this would be error-prone, because we also need to manipulate weights.

**Comment by lahabana** [comment]
Yes! I think that's what I've thinking for a while that there could be a cluster cache and these confs are more controller

**Comment by slonka** [comment]
I think @Automaat recently did some changes to MeshContext, has anything changed around this?

**Comment by slonka** [comment]
> triage: let's consider this for zone proxies rework, we will figure out if we need to touch this part of the codebase and close it if not needed

**Comment by lukidzi** [comment]
The zone proxy rework (MADR-095) embeds zone ingress/egress listeners directly into Dataplane resources. The new path uses `BuildDataplaneZoneIngressEndpointMap` and `BuildDataplaneZoneEgressEndpointMap` which are purpose-built and don't have the double-build problem described here.(https://github.com/kumahq/kuma/pull/15870)
As standalone `ZoneIngressResource`/`ZoneEgressResource` proxies are superseded by this approach, the `BuildIngressEndpointMap` path this issue targets will eventually be removed. Implementing the optimization now would add weight-normalization complexity to code that's on its way out.

Closing as superseded by the zone proxy rework
